### PR TITLE
chore: usage tracking channel

### PR DESCRIPTION
**Pull request by hoangharry**
**What does this PR do?**

-   [x] usage tracking channel

**Media (Loom or gif)**

-   Insert media here (if applicable)
Screenshot 2023-03-06 at 16 51 55

### CODE REVIEW

File: src/logger/usage-tracking.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    try {
+      const guild = client.guilds.cache.get(MOCHI_GUILD_ID)
+      if (guild) {
+        this.logChannel = guild.channels.cache.get(
+          USAGE_TRACKING_CHANNEL_ID
+        ) as TextChannel
+      }
+    } catch (e: any) {
+      logger.error(e)
+    }
```
Comment (anhnh12) [inline src/logger/usage-tracking.ts:24]: remove try-catch, can use `wrapError()`. remember to re-run and test

File: src/logger/usage-tracking.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  CommandInteraction,
+  Message,
+  MessageEmbed,
+  TextChannel,
+} from "discord.js"
+import { MOCHI_GUILD_ID, USAGE_TRACKING_CHANNEL_ID } from "env"
+import { logger } from "logger"
+
+export class UsageTrackingLogger {
+  logChannel: TextChannel | null = null
```
Comment (anhnh12) [inline src/logger/usage-tracking.ts:12]: private

File: src/listeners/discord/ready.ts [hunk 3]
```diff
@@ -22,6 +23,7 @@ const event: DiscordEvent<"ready"> = {
       if (!client.user) return
       logger.info(`Bot [${client.user.username}] is ready`)
       ChannelLogger.ready(client)
+      UsageTrackingLogger.ready(client)
```
Comment (anhnh12) [inline src/listeners/discord/ready.ts:26]: any specific reason for this?
why don't put this inside `UsageTrackingLogger` constructor()

**Review by anhnh12 — CHANGES_REQUESTED** [review]

File: src/listeners/discord/ready.ts (hunk 3 above)
Comment (hoangharry) [reply src/listeners/discord/ready.ts:26]: put here as the same as ChannelLogger for the same pattern

File: src/logger/usage-tracker.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+      if (msg instanceof Message) {
+        content = msg.content
+        authorName = msg.author.username
+        authorAvatar = msg.author.avatarURL() ?? ""
+      }
+      if (msg instanceof CommandInteraction) {
+        content = `**Slash Command:** ${msg.commandName}`
+        authorName = msg.user.username
+        authorAvatar = msg.user.avatarURL() ?? ""
+      }
```
Comment (anhnh12) [inline src/logger/usage-tracker.ts:34]:
```suggestion
    const author = getAuthor(msg)
    const isMessage = msg instanceof Message
    const content = isMessage ? msg.content : `**Slash Command:** ${msg.commandName}`
```

File: src/logger/usage-tracker.ts [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+      const embed = composeEmbedMessage(null, {
+        description: content,
+        author: [msg.guild?.name ?? "", msg.guild?.bannerURL() ?? ""],
+      })
+        .setTimestamp()
```
Comment (anhnh12) [inline src/logger/usage-tracker.ts:39]: no need

File: src/logger/usage-tracker.ts [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+        .setFooter({ text: authorName, iconURL: authorAvatar })
```
Comment (anhnh12) [inline src/logger/usage-tracker.ts:40]: remove.
instead pass `originalMsgAuthor` to `composeEmbedMessage()`

File: src/logger/usage-tracker.ts [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+        .setAuthor({
+          name: msg.guild?.name ?? "",
+          iconURL: msg.guild?.bannerURL() ?? "",
+        })
```
Comment (anhnh12) [inline src/logger/usage-tracker.ts:44]: pass `author` to `composeEmbedMessage`

**Review by anhnh12 — CHANGES_REQUESTED** [review]

File: src/logger/usage-tracker.ts (hunk 6 above)
Comment (hoangharry) [reply src/logger/usage-tracker.ts:40]: pass `originalMsgAuthor` will get unnecessary footer `Type /feedback to report`

**Review by anhnh12 — APPROVED** [review]

**Pull request merged**